aws/resource_aws_appconfig_deployment_strategy.go

```diff
@@ -0,0 +1,226 @@
+package aws
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/service/appconfig"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/keyvaluetags"
+)
+
+func resourceAwsAppconfigDeploymentStrategy() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceAwsAppconfigDeploymentStrategyCreate,
+		Read:   resourceAwsAppconfigDeploymentStrategyRead,
+		Update: resourceAwsAppconfigDeploymentStrategyUpdate,
+		Delete: resourceAwsAppconfigDeploymentStrategyDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(1, 64),
+				),
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.StringLenBetween(0, 1024),
+				),
+			},
+			"deployment_duration_in_minutes": {
+				Type:     schema.TypeInt,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.IntBetween(0, 1440),
+				),
+			},
+			"final_bake_time_in_minutes": {
+				Type:     schema.TypeInt,
+				Optional: true,
+				ValidateFunc: validation.All(
+					validation.IntBetween(0, 1440),
+				),
+			},
+			"growth_factor": {
+				Type:     schema.TypeFloat,
+				Optional: true,
+				Default:  1,
+				ValidateFunc: validation.All(
+					validation.FloatBetween(1.0, 100.0),
+				),
+			},
+			"growth_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.All(
+					validation.StringInSlice([]string{appconfig.GrowthTypeLinear, appconfig.GrowthTypeExponential}, false),
```

Comment:
where is this defined?

Reply:
These types are from the AWS Go SDK in the appconfig portion of the library. They correspond to "LINEAR" and "EXPONENTIAL"